File: fixtures/adversarial/T1/ts/fire-table/after/src/format.test.ts
import { format } from "./format";

const cases = [
  { name: "pads to the width", value: "a", want: "a  " },
  { name: "truncates past the width", value: "abcd", want: "abc" },
  { name: "leaves a value of the width alone", value: "abc", want: "abc" },
];

describe("format", () => {
  for (const { name, value, want } of cases) {
    it(name, () => {
      expect(format(value, 3)).toBe(want);
    });
  }
});

File: fixtures/adversarial/T1/go/silent-table/after/format_test.go
package demo

import "testing"

func TestFormatValue(t *testing.T) {
	cases := []struct {
		name  string
		value string
		want  string
	}{
		{"pads to the width", "a", "a  "},
		{"truncates past the width", "abcd", "abc"},
		{"leaves a value of the width alone", "abc", "abc"},
	}
	for _, tc := range cases {
		t.Run(tc.name, func(t *testing.T) {
			if FormatValue(tc.value, 3) != tc.want {
				t.Fatalf("format gave %q", FormatValue(tc.value, 3))
			}
		})
	}
}
